streampipes-extensions/streampipes-processors-transformation-jvm/src/main/java/org/apache/streampipes/processors/transformation/jvm/processor/round/RoundProcessor.java

```diff
@@ -0,0 +1,85 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.streampipes.processors.transformation.jvm.processor.round;
+
+import org.apache.streampipes.commons.exceptions.SpRuntimeException;
+import org.apache.streampipes.extensions.api.pe.context.EventProcessorRuntimeContext;
+import org.apache.streampipes.extensions.api.pe.routing.SpOutputCollector;
+import org.apache.streampipes.model.DataProcessorType;
+import org.apache.streampipes.model.graph.DataProcessorDescription;
+import org.apache.streampipes.model.runtime.Event;
+import org.apache.streampipes.model.schema.PropertyScope;
+import org.apache.streampipes.sdk.builder.ProcessingElementBuilder;
+import org.apache.streampipes.sdk.builder.StreamRequirementsBuilder;
+import org.apache.streampipes.sdk.helpers.EpRequirements;
+import org.apache.streampipes.sdk.helpers.Labels;
+import org.apache.streampipes.sdk.helpers.Locales;
+import org.apache.streampipes.sdk.helpers.OutputStrategies;
+import org.apache.streampipes.sdk.utils.Assets;
+import org.apache.streampipes.wrapper.params.compat.ProcessorParams;
+import org.apache.streampipes.wrapper.standalone.StreamPipesDataProcessor;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+
+public class RoundProcessor extends StreamPipesDataProcessor {
+  private List<String> fieldsToBeRounded;
+  private int numDigits;
+
+  private static final String FIELDS = "fields";
+  private static final String NUM_DIGITS = "num-digits";
+
+  @Override
+  public DataProcessorDescription declareModel() {
+    return ProcessingElementBuilder
+        .create("org.apache.streampipes.processors.transformation.jvm.round")
+        .category(DataProcessorType.TRANSFORM)
+        .withLocales(Locales.EN)
+        .withAssets(Assets.DOCUMENTATION, Assets.ICON)
+        .requiredStream(StreamRequirementsBuilder
+            .create()
+            .requiredPropertyWithNaryMapping(EpRequirements.numberReq(), Labels.withId(FIELDS), PropertyScope.NONE)
+            .build())
+        .requiredIntegerParameter(Labels.withId(NUM_DIGITS))
+        .outputStrategy(OutputStrategies.keep())
+        .build();
+  }
+
+  @Override
+  public void onInvocation(ProcessorParams parameters, SpOutputCollector spOutputCollector,
+                           EventProcessorRuntimeContext runtimeContext) throws SpRuntimeException {
+    fieldsToBeRounded = parameters.extractor().mappingPropertyValues(FIELDS);
+    numDigits = parameters.extractor().singleValueParameter(NUM_DIGITS, Integer.class);
+  }
+
+  @Override
+  public void onEvent(Event event, SpOutputCollector collector) throws SpRuntimeException {
+    for (String fieldToBeRounded : fieldsToBeRounded) {
+      double value = event.getFieldBySelector(fieldToBeRounded).getAsPrimitive().getAsDouble();
+      double roundedValue = BigDecimal.valueOf(value).setScale(numDigits, RoundingMode.HALF_UP).doubleValue();
```

Comment:
Could you please make the rounding mode a user configurable input too?

Reply:
For sure, I will support all the rounding modes specified [here](https://docs.oracle.com/javase/8/docs/api/java/math/RoundingMode.html). Should we also support custom number to round up (e.g. 3 to round up means 1.3 will be rounded to 2)? I believe this is somewhat unnecessary because the user can just apply some mathematical transformations to achieve this. What do you think?

Reply:
For the icon, I am really really really bad at hand/digital drawing... 🤣 It would be great if you can help come up with the icon.

Reply:
> Should we also support custom number to round up (e.g. 3 to round up means 1.3 will be rounded to 2)? 
Sorry, I don't really get your point. Can you please rephrase it? 

No, worries. I'll come up with one 🙂

Reply:
> > Should we also support custom number to round up (e.g. 3 to round up means 1.3 will be rounded to 2)?
> 
> Sorry, I don't really get your point. Can you please rephrase it?
> 
> > For the icon, I am really really really bad at hand/digital drawing... 🤣 It would be great if you can help come up with the icon.
> 
> No, worries. I'll come up with one 🙂

Sorry for my confusing phrasing. I basically mean I believe only [these modes](https://docs.oracle.com/javase/8/docs/api/java/math/RoundingMode.html) need to be supported. Is there any other rounding mode that you think users will want to use?

Reply:
Ah yes, I think they are perfectly fine 👍🏼